src/parser/demonhunter/havoc/modules/spells/BladeDance.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPECS from 'game/SPECS';
+import Events, { DamageEvent, FightEndEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import SpellLink from 'common/SpellLink';
+
+//For Blade dance and Death Sweep
+class BladeDance extends Analyzer {
+  get suggestionThresholds() {
+    return {
+      actual: this.badCast,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 1,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  badCast = 0;
+  hitCount = 0;
+  hitTimeStamp: number = 0;
+  firstHitTimeStamp: number = 0;
+  strikeTime: number = 1000;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.spec === SPECS.HAVOC_DEMON_HUNTER;
+    if (!this.active || this.selectedCombatant.hasTalent(SPELLS.TRAIL_OF_RUIN_TALENT) || this.selectedCombatant.hasTalent(SPELLS.FIRST_BLOOD_TALENT)) {
+        return;
+    }
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP_DAMAGE), this.onDamage);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.BLADE_DANCE_DAMAGE), this.onDamage);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.BLADE_DANCE_DAMAGE_LAST_HIT), this.onDamage);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP_DAMAGE_LAST_HIT), this.onDamage);
+    this.addEventListener(Events.fightend, this.onFightEnd);
+  }
+
+  onDamage(event: DamageEvent) {
+    //Function both for Blade Dance and Death Sweep
+    //less than 5 hits = single target, bad cast.
+    this.hitTimeStamp = event.timestamp;
+    if (event.timestamp > this.firstHitTimeStamp + this.strikeTime){
+      //New Strike
+      if(this.hitCount < 5 && this.hitCount > 1) {
+        //chech if last strike was bad
+        this.badCast += 1;
+      }
+      this.firstHitTimeStamp = this.hitTimeStamp //Timestamp for first hit in strike
+      this.hitCount = 0;
+    }
+    this.hitCount += 1;
+  }
+
+  onFightEnd(event: FightEndEvent){
+    if (this.hitCount < 5){
```

Comment:
Is it accurate to say 
If the last cast of the fight is blade dance but the fight finished before blade dance could finish it was a bad cast?
if so then cool if not then this is going to inflate your bad casts by 1

Reply:
I would say if you do blade dance in the end of the fight and it doesnt hit enough, its a bad cast because you have other spells available to you that could do more damage. If it would be because you need 5 more fury for your main non-cd abiliity, melee + demons bite would do more damage in the same amount of time.

Reply:
lit